hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerSet.java

```diff
@@ -72,6 +75,9 @@ public boolean addContainer(Container<?> container) throws
       }
       // wish we could have done this from ContainerData.setState
       container.getContainerData().commitSpace();
+      if (container.getContainerData().getState() == RECOVERING) {
```

Comment:
As suggested below in tests, we can use clock.millis() for time.

Reply:
Also we can set the time currentTime + <Configured Interval>, so that we can just check whether time elapsed or not when checking. We don;t need to do + op every time.